src/FixedSizeArrays.jl

```diff
@@ -38,6 +38,24 @@ end
 
 Base.isassigned(a::FixedSizeArray, i::Int) = isassigned(a.mem, i)
 
+# safe product of a tuple of integers, for calculating dimensions size
+
+checked_dims_impl(a::Int, ::Tuple{}) = a
+function checked_dims_impl(a::Int, t::Tuple{Int,Vararg{Int,N}}) where {N}
```

Comment:
You might want to copy the implementation from Julia base, since this one lacks handling of negative axes, which can lead to odd behaviors